homeassistant/components/kraken/__init__.py

```diff
@@ -0,0 +1,152 @@
+"""The kraken integration."""
+from __future__ import annotations
+
+import asyncio
+from datetime import timedelta
+import logging
+
+import async_timeout
+import krakenex
+import pykrakenapi
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_SCAN_INTERVAL
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.dispatcher import async_dispatcher_send
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import (
+    CONF_TRACKED_ASSET_PAIRS,
+    DEFAULT_SCAN_INTERVAL,
+    DEFAULT_TRACKED_ASSET_PAIR,
+    DISPATCH_CONFIG_UPDATED,
+    DOMAIN,
+)
+from .utils import get_tradable_asset_pairs
+
+PLATFORMS = ["sensor"]
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass: HomeAssistant, config_entry: ConfigEntry) -> bool:
+    """Set up kraken from a config entry."""
+    kraken_data = KrakenData(hass, config_entry)
+    await kraken_data.async_setup()
+    hass.data[DOMAIN] = kraken_data
+    config_entry.add_update_listener(async_options_updated)
```

Comment:
We should save the return value here and use it to unsubscribe the listener on entry unload.

Reply:
```suggestion
    config_entry.async_on_unload(config_entry.add_update_listener(async_options_updated))
```